scipy/ndimage/_measurements.py

```diff
@@ -507,6 +508,8 @@ def labeled_comprehension(input, labels, index, func, out_dtype, default,
         else:
             return func(input.ravel(), positions.ravel())
 
+    labels = np.asarray(labels)
```

Comment:
was the need for these extra `np.asarray`'s caught by the new testing?

Reply:
Yes. Otherwise e.g. torch arrays end up in numpy-specific ndimage code which does not end well.